utils/funcexp/functor_bool.h

```diff
@@ -241,6 +242,94 @@ class Func_isnull : public Func_Bool
 };
 
 
+/** @brief Func_Truth class
+  */
+class Func_Truth : public Func_Bool
+{
+public:
+    Func_Truth(const std::string& funcName, bool a_value, bool a_affirmative) :
+        Func_Bool(funcName), value(a_value), affirmative(a_affirmative) {}
+
+    virtual ~Func_Truth() {}
+
+    execplan::CalpontSystemCatalog::ColType operationType(FunctionParm& fp, execplan::CalpontSystemCatalog::ColType& resultType)
+    {
+        assert (fp.size() == 1);
+        return fp[0]->data()->resultType();
```

Comment:
Shouldn't this return the type that corresponds to a boolean?

Reply:
I am honestly not sure how operationType() is even used in the first place. I used the same code here as in Func_isnull::operationType